perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -151,7 +151,9 @@ public PerunSession getPerunSession(PerunPrincipal principal, PerunClient client
 						usersManagerBl.updateUserExtSource(internalSession, ues);
 					}
 					// Update last access for userExtSource
-					usersManagerBl.updateUserExtSourceLastAccess(internalSession, ues);
+					if(!BeansUtils.isPerunReadOnly()) {
```

Comment:
I would prefer to skip all update methods in read-only mode, so that we support also other ways of authentication and basically we allow reading from replica.